sympy/tensor/array/arrayop.py

```diff
@@ -208,3 +208,38 @@ def derive_by_array(expr, dx):
             return ImmutableDenseNDimArray([expr.diff(i) for i in dx], dx.shape)
         else:
             return diff(expr, dx)
+
+
+def permutedims(expr, perm):
+    """
+    Permutes the indices of an array.
+
+    Parameter specifies the permutation of the indices.
+    """
+    if not isinstance(expr, NDimArray):
+        raise TypeError("expression has to be an N-dim array")
+
+    perm = tuple(perm)
```

Comment:
Probably it should be better to use `sympy.combinatorics.Permutation` and replace lines 222, 225-228 with
```python
from sympy.combinatorics import Permutation
perm = Permutation(perm)
inverse = ~perm
```